lib/setup/setupUpgrade.js

```diff
@@ -67,24 +67,38 @@ function Upgrade(options) {
         let adapters;
         if (dependencies) {
             // if like [{"js-controller": ">=0.10.1"}]
-            if (dependencies instanceof Array) {
+            if (Array.isArray(dependencies)) {
                 adapters = {};
                 dependencies.filter(a => typeof dependencies[a] !== 'string')
-                    .forEach(rule => Object.keys(rule).filter(adapter => !adapters[adapter]).forEach(adapter => adapters[adapter] = rule[adapter]));
-            } // if like {"js-controller": ">=0.10.1", "admin": ">=0.5.6"}
-            else {
+                    .forEach(rule => {
+                        if (typeof rule === 'string') {
+                            // No version given, all are okay
+                            adapters[rule] = '*';
+                        } else if (tools.isObject(rule)) {
+                            Object.keys(rule).filter(adapter => !adapters[adapter]).forEach(adapter => adapters[adapter] = rule[adapter]);
+                        }
+                    });
+            } else {
```

Comment:
Also here ...maybe that that it is really an object and that not someone put in a string here? ;-)